pkgs/development/python-modules/scikit-learn/default.nix

```diff
@@ -61,15 +61,27 @@ buildPythonPackage rec {
   LC_ALL="en_US.UTF-8";
 
   preBuild = ''
-    export SKLEARN_BUILD_PARALLEL=$NIX_BUILD_CORES
+    export SKLEARN_BUILD_PARALLEL=1
   '';
 
   doCheck = !stdenv.isAarch64;
 
-  # Skip test_feature_importance_regression - does web fetch
-  disabledTests = [ "test_feature_importance_regression" ];
+  disabledTests = [
+    # Skip test_feature_importance_regression - does web fetch
+    "test_feature_importance_regression"
+    # NuSVC memmap tests causes segmentation faults in certain environments
+    # (e.g. Hydra Darwin machines) related to a long-standing joblib issue
+    # (https://github.com/joblib/joblib/issues/563). See also:
+    # https://github.com/scikit-learn/scikit-learn/issues/17582
+    "(NuSVC and memmap)"
+  ];
```

Comment:
```suggestion
  disabledTests = [
    # Skip test_feature_importance_regression - does web fetch
    "test_feature_importance_regression"

    # failing on macos
    "check_regressors_train"
    "check_classifiers_train"
    "xfail_ignored_in_check_estimator"

    # NuSVC memmap tests causes segmentation faults in certain environments
    # (e.g. Hydra Darwin machines) related to a long-standing joblib issue
    # (https://github.com/joblib/joblib/issues/563). See also:
    # https://github.com/scikit-learn/scikit-learn/issues/17582
    "NuSVC"
    "memmap"
  ];
```

Reply:
`(NuSVC and memmap)` didn't work because pytestCheckHook splits words separated by spaces and interlaces them with `and not`.

Reply:
The intent here was to only disable tests matching both `NuSVC AND memmap`, but not `NuSVC OR memmap`. I don't think this is possible to achieve through disabledTests.

I originally had this in pytestFlagsArray, but then was asked to change it: https://github.com/NixOS/nixpkgs/pull/122687#discussion_r634263657

Reply:
@SuperSandro2000 I guess this should just go into pytestFlagsArray then? Or do you know a better way?

Reply:
If the old way works (`pytestFlagsArray`) and disables the right stuff, let's go back to that. The request was well-intentioned, but this is a special case / complex issue, so the general advice need not necessarily apply here. Sorry for all of us yanking you back and forth, @nbren12. You've been a good sport :P

Reply:
I just restored the old way, and it works locally for me:
```
$ nix-build -A python39Packages.scikit-learn . -j 4
/nix/store/g0q4fmry9kg1cc9ia1zqan0qj2ys5zji-python3.9-scikit-learn-0.24.1
```

Reply:
Seems like pytest doesn't accept multiple `-k` flags. Therefore you cannot disable tests via `pytestFlagsArray` and `disabledTests` at the same time.